File: HashTable/3.Group Anagrams (0049)/Solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        HashMap<String, List<String>> hashMap = new HashMap<>();

        for(String word : strs){
            char[] chars = word.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);

            if(!hashMap.containsKey(key)){
                hashMap.put(key, new ArrayList<>());
            }

            hashMap.get(key).add(word);  

            // if(!hashMap.containsKey(key)){
            //     List<String> value = new ArrayList<>();
            //     value.add(word);
            //     hashMap.put(key, value);
            // }else{
            //     hashMap.get(key).add(word);
            // }                        
        }

        return new ArrayList<>(hashMap.values());
    }
}

File: HashTable/3.Group Anagrams (0049)/Solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = dict()
        for word in strs:
            char_array = list(word)
            char_array.sort()
            key = ''.join(char_array)

            if dic.get(key) is None:
                dic[key] = list()

            dic[key].append(word)
        return list(dic.values())           
